cirq/google/engine/engine_program.py

```diff
@@ -387,28 +387,45 @@ def remove_labels(self, keys: List[str]) -> 'EngineProgram':
             self.project_id, self.program_id, keys)
         return self
 
-    def get_circuit(self) -> 'Circuit':
+    def get_circuit(self, program_num: Optional[int] = None) -> 'Circuit':
         """Returns the cirq Circuit for the Quantum Engine program. This is only
         supported if the program was created with the V2 protos.
 
+        Args:
+            program_num: if this is a batch program, the index of the circuit in
+                the batch.  This argument is zero-indexed.
+
         Returns:
             The program's cirq Circuit.
         """
         if not self._program or not self._program.HasField('code'):
             self._program = self.context.client.get_program(
                 self.project_id, self.program_id, True)
-        return self._deserialize_program(self._program.code)
+        return self._deserialize_program(self._program.code, program_num)
 
     @staticmethod
-    def _deserialize_program(code: qtypes.any_pb2.Any) -> 'Circuit':
+    def _deserialize_program(code: qtypes.any_pb2.Any,
+                             program_num: Optional[int] = None) -> 'Circuit':
         import cirq.google.engine.engine as engine_base
         code_type = code.type_url[len(engine_base.TYPE_PREFIX):]
         if (code_type == 'cirq.google.api.v1.Program' or
                 code_type == 'cirq.api.google.v1.Program'):
             raise ValueError('deserializing a v1 Program is not supported')
+        program = None
         if (code_type == 'cirq.google.api.v2.Program' or
                 code_type == 'cirq.api.google.v2.Program'):
             program = v2.program_pb2.Program.FromString(code.value)
+        if code_type == 'cirq.google.api.v2.BatchProgram':
+            if program_num is None:
+                raise ValueError('A program number must be specified when '
+                                 'deserializing a Batch Program')
+            batch = v2.batch_pb2.BatchProgram.FromString(code.value)
+            if program_num >= len(batch.programs):
```

Comment:
The
if math.abs(program_num) >= len(batch.programs)
would be more appropriate

Reply:
Done.